File: sprint4/C#/lesson1code4.cs
using System;
using System.Collections.Generic;

public struct Pair {
    public string key;
    public string value;
}

public class Map {
    private List<Pair> pairs = new List<Pair>();

    public string get(string key) {
        foreach (var pair in pairs) {
            if (pair.key == key) { // пара с указанным ключом найдена
                return pair.value;
            }
        }
        return ""; // пара с указанным ключом не найдена
    }

    public void set(string key, string value) {
        foreach (var pair in pairs) {
            if (pair.key == key) { // пара с указанным ключом найдена
                // обновить значение в найденной паре
                pair.value = value;
                return;
            }
        }
        // пара с заданным ключом не найдена
        pairs.Add(new Pair{key = key, value = value});
    }
}

File: sprint4/By lessons/lesson1code4/lesson1code4.py
class Pair:
    def __init__(self, key, value):
        self.key = key
        self.value = value

class Map:
    def __init__(self):
        self.pairs = []

    def get(self, key):
        for pair in self.pairs:
            if pair.key == key:
                return pair.value
        return None

    def set(self, key, value):
        for pair in self.pairs:
            if pair.key == key:
                pair.value = value
                return
        new_pair = Pair(key, value)
        self.pairs.append(new_pair)
